cmd/juju/cloud/add_test.go

```diff
@@ -148,83 +182,146 @@ func (*addSuite) TestAddBadCloudName(c *gc.C) {
 
 func (*addSuite) TestAddExisting(c *gc.C) {
 	fake := newFakeCloudMetadataStore()
-	fake.Call("ParseCloudMetadataFile", "fake.yaml").Returns(homestackMetadata(), nil)
+	mockStore := &mockCloudeMetadataStore{}
```

Comment:
it's not clear to me why there's now both a "fake" and "mock" store. and the mock store is only used by the test code? the mock store doesn't seem to be needed